sdks/java/io/hbase/src/test/java/org/apache/beam/sdk/io/hbase/HBaseIOTest.java

```diff
@@ -0,0 +1,449 @@
+import org.apache.beam.sdk.testing.PAssert;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.transforms.Count;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.transforms.display.DisplayData;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.commons.lang.StringUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.HBaseConfiguration;
+import org.apache.hadoop.hbase.HBaseTestingUtility;
+import org.apache.hadoop.hbase.HColumnDescriptor;
+import org.apache.hadoop.hbase.HConstants;
+import org.apache.hadoop.hbase.HTableDescriptor;
+import org.apache.hadoop.hbase.TableName;
+import org.apache.hadoop.hbase.client.BufferedMutator;
+import org.apache.hadoop.hbase.client.Connection;
+import org.apache.hadoop.hbase.client.ConnectionFactory;
+import org.apache.hadoop.hbase.client.HBaseAdmin;
+import org.apache.hadoop.hbase.client.Mutation;
+import org.apache.hadoop.hbase.client.Put;
+import org.apache.hadoop.hbase.client.Result;
+import org.apache.hadoop.hbase.client.ResultScanner;
+import org.apache.hadoop.hbase.client.Scan;
+import org.apache.hadoop.hbase.client.Table;
+import org.apache.hadoop.hbase.filter.CompareFilter;
+import org.apache.hadoop.hbase.filter.Filter;
+import org.apache.hadoop.hbase.filter.RegexStringComparator;
+import org.apache.hadoop.hbase.filter.RowFilter;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.hamcrest.Matchers;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/**
+ * Test HBaseIO.
+ */
+@RunWith(JUnit4.class)
+public class HBaseIOTest {
+    @Rule public ExpectedException thrown = ExpectedException.none();
+    @Rule public transient ExpectedLogs logged = ExpectedLogs.none(HBaseIO.class);
+
+    private static HBaseTestingUtility htu;
+    private static HBaseAdmin admin;
+
+    private static Configuration conf = HBaseConfiguration.create();
+    private static final byte[] COLUMN_FAMILY = Bytes.toBytes("info");
+    private static final byte[] COLUMN_NAME = Bytes.toBytes("name");
+    private static final byte[] COLUMN_EMAIL = Bytes.toBytes("email");
+
+    @BeforeClass
+    public static void beforeClass() throws Exception {
+        try {
+            conf.setInt(HConstants.HBASE_CLIENT_RETRIES_NUMBER, 1);
+            htu = new HBaseTestingUtility(conf);
+            htu.startMiniCluster(1, 4);
+            admin = htu.getHBaseAdmin();
+
+            CoderRegistry cr = TestPipeline.create().getCoderRegistry();
+            cr.registerCoder(Mutation.class, HBaseMutationCoder.class);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @AfterClass
+    public static void afterClass() throws Exception {
+        admin.close();
+        htu.shutdownMiniCluster();
```

Comment:
these should always look like `if (admin != null) { admin.close(); admin = null; }` that will make your tests more resilient.